### ISSUE TITLE
add support for Registration Reject (AMF to UE)

### ISSUE THREAD

**Issue by sureshgorijavolu** [magma/magma#6540]

**Comment by Chandhu-Wavelabs** [comment]
PR #8190 raised for Registration Reject, but this PR is dependent on PR #7565, until #7565 merged this PR is in draft mode.

Present 166 files are reflecting in PR but once #7565 merged we rebase our PR with latest master, then only delta will be reflected. 

In the following link we can see the delta related to Registration Reject.
https://github.com/magma/magma/pull/8190/commits/5cc56032569a50bfff2c87269fefd2646c57f214

We can use the following commit ID to cherry pick the delta related to Registration Reject.
Commit-ID : 5cc56032569a50bfff2c87269fefd2646c57f214

**Comment by Chandhu-Wavelabs** [comment]
Old PR #8190 closed and new PR #8489 raised against to master branch. Need to be reviewed.

**Comment by Chandhu-Wavelabs** [comment]
Merged
